dash-renderer/src/TreeContainer.js

```diff
@@ -102,49 +109,48 @@ class TreeContainer extends Component {
 
     setProps(newProps) {
         const {
-            _dashprivate_dependencies,
+            _dashprivate_graphs,
             _dashprivate_dispatch,
             _dashprivate_path,
             _dashprivate_layout,
         } = this.props;
 
-        const id = this.getLayoutProps().id;
-
-        // Identify the modified props that are required for callbacks
-        const watchedKeys = filter(
-            key =>
-                _dashprivate_dependencies &&
-                _dashprivate_dependencies.find(
-                    dependency =>
-                        dependency.inputs.find(
-                            input => input.id === id && input.property === key
-                        ) ||
-                        dependency.state.find(
-                            state => state.id === id && state.property === key
-                        )
-                )
-        )(keysIn(newProps));
-
-        // setProps here is triggered by the UI - record these changes
-        // for persistence
-        recordUiEdit(_dashprivate_layout, newProps, _dashprivate_dispatch);
-
-        // Always update this component's props
-        _dashprivate_dispatch(
-            updateProps({
-                props: newProps,
-                itempath: _dashprivate_path,
-            })
+        const oldProps = this.getLayoutProps();
+        const {id} = oldProps;
+        const changedProps = pickBy(
+            (val, key) => !equals(val, oldProps[key]),
+            newProps
         );
+        const changedKeys = keys(changedProps);
+        if (changedKeys.length) {
```

Comment:
Since none of this is time critical, maybe we can keep with the Ramda abstraction by using `isEmpty` instead of `array.length`

Reply:
yep, that's a little simpler -> https://github.com/plotly/dash/pull/1103/commits/14b7e3ca90b0e33e63ef9ebc0178503644848aab